File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        dupes = {}
        
        for num in nums:
            if num not in dupes:
                dupes[num] = 1
            else:
                print(dupes)
                return True
            
        return False
        

File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let seen = {};

    for (let num of nums) {
        if (seen[num]) {
            return true;
        }

        seen[num] = true;
    }

    return false;
};